### PR TITLE
[MicrosoftManagementActivity] updated fetch start_time

### PR DESCRIPTION
**Pull request by michal-dagan**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/XSUP-24583)

## Description
Fixed an issue where the start time for fetching incidents from the last retrieval was not parsed correctly.

From Microsoft docs: https://learn.microsoft.com/en-us/office/office-365-management-api/office-365-management-activity-api-reference#list-available-content: `Both must be specified (or both omitted) and they must be no more than 24 hours apart, with the start time no more than 7 days in the past. By default, if startTime and endTime are omitted, then the content available in the last 24 hours is returned.`

### CODE REVIEW

File: Packs/MicrosoftManagementActivity/Integrations/MicrosoftManagementActivity/MicrosoftManagementActivity.py [hunk 1]
```diff
@@ -457,11 +457,14 @@ def get_fetch_start_and_end_time(last_run, first_fetch_datetime):
...
         fetch_start_datetime = datetime.strptime(last_fetch, DATE_FORMAT)
 
+    # the start time must be no more than 7 days in the past
+    fetch_start_datetime = max(fetch_start_datetime, dateparser.parse("7 days ago"))
     fetch_end_datetime = get_fetch_end_time_based_on_start_time(fetch_start_datetime)
 
     # The API expects strings of format YYYY:DD:MMTHH:MM:SS
     fetch_start_time_str = fetch_start_datetime.strftime(DATE_FORMAT)
     fetch_end_time_str = fetch_end_datetime.strftime(DATE_FORMAT)
+    demisto.debug(f"get_fetch_start_and_end_time: {fetch_start_time_str=}, {fetch_end_time_str=}")
```
Comment (bziser) [inline Packs/MicrosoftManagementActivity/Integrations/MicrosoftManagementActivity/MicrosoftManagementActivity.py:467]: great that you added logging!

**Review by bziser — COMMENTED** [review]
good job

**Review by bziser — APPROVED** [review]

**Comment by ShirleyDenkberg** [comment]
@bziser Doc review completed.  Looks good.

File: Packs/MicrosoftManagementActivity/Integrations/MicrosoftManagementActivity/README.md [hunk 2]
```diff
@@ -308,4 +308,5 @@ There is no context output for this command.
 
 
 ## Additional Information
-Record types to fetch from should be set with numerical values from the [Microsoft documentation](https://docs.microsoft.com/en-us/office/office-365-management-api/office-365-management-activity-api-schema#auditlogrecordtype). For example, in order to fetch events of type **MailSubmission**, the value **29** should be set.
+- Record types to fetch from should be set with numerical values from the [Microsoft documentation](https://docs.microsoft.com/en-us/office/office-365-management-api/office-365-management-activity-api-schema#auditlogrecordtype). For example, in order to fetch events of type **MailSubmission**, the value **29** should be set.
+- When fetching, note that the API only supports a start time that is no more than 7 days in the past. In cases where the last fetch timestamp exceeds this limitation, the integration will automatically default to fetching data from 7 days ago.
```
Comment (bziser) [inline Packs/MicrosoftManagementActivity/Integrations/MicrosoftManagementActivity/README.md:312]:
```suggestion
-  Note that the API only supports start times up to 7 days in the past when fetching. If the last fetch timestamp exceeds this limit, the integration automatically fetches data from 7 days ago.
```

File: Packs/MicrosoftManagementActivity/Integrations/MicrosoftManagementActivity/README.md [hunk 3]
```diff
@@ -308,4 +308,5 @@ There is no context output for this command.
 
 
 ## Additional Information
-Record types to fetch from should be set with numerical values from the [Microsoft documentation](https://docs.microsoft.com/en-us/office/office-365-management-api/office-365-management-activity-api-schema#auditlogrecordtype). For example, in order to fetch events of type **MailSubmission**, the value **29** should be set.
+- Record types to fetch from should be set with numerical values from the [Microsoft documentation](https://docs.microsoft.com/en-us/office/office-365-management-api/office-365-management-activity-api-schema#auditlogrecordtype). For example, in order to fetch events of type **MailSubmission**, the value **29** should be set.
+-  Note that the API only supports start times up to 7 days in the past when fetching. If the last fetch timestamp exceeds this limit, the integration automatically fetches data from 7 days ago.
```
Comment (michal-dagan) [inline Packs/MicrosoftManagementActivity/Integrations/MicrosoftManagementActivity/README.md:312]:
```suggestion
- Note that the API only supports start times up to 7 days in the past when fetching. If the last fetch timestamp exceeds this limit, the integration automatically fetches data from 7 days ago.
```

**Pull request merged**